optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/domain/TestdataEntity.java

```diff
@@ -37,7 +37,7 @@
         return entityDescriptor.getGenuineVariableDescriptor("value");
     }
 
-    private TestdataValue value;
+    public TestdataValue value;
```

Comment:
I have to say I absolutely hate this. Public fields are an antipattern, and we are now proliferating it throughout the tests.
There has to be a better way of testing this.

Reply:
I also hate it. If there were some way to shallow clone a Java Object (via a method or bytecode instruction), then in theory we could do the planning cloning using only the member accessors (in which case the old restrictions apply: you can use Gizmo if and only if all planning annotations are on public members).

Reply:
I have accepted the limitation that Gizmo would only be useful in Quarkus - we can't change that. But can we do anything about the tests?

If we only run the Gizmo tests in Quarkus, we do not have to make everything public. Alternatively, we could introduce some special Testdata domain that is only use for Gizmo, and keep the rest of the domain as is.

Reply:
The main issue is sharing the test cases between the two implementations of SolutionCloners (FieldAccessingSolutionCloner and GizmoSolutionCloner). It might be possible if we import the test-jar of OptaPlanner Core into OptaPlanner Quarkus. But then we don't have testing for the plain java case (it can work in Java, just with limitations).

Reply:
Or we give up on running the same tests in both environments.
Sonds like copy-pasting the test and specializing it for Gizmo may end up being the best solution.

Reply:
I think there may be a different solution, but it is a hail mary pass. The only reason why we can not use Gizmo with non-public fields is that we can not redefine classes. They are already loaded!

Well, they aren't, if you use a different classloader. :-) So if we mess around with classloaders and run the test only with the correct classloader with the Gizmo-instrumented classes, it should work.

However, once you start doing black magic with classloaders, you realize you are probably doing something you shouldn't. It tends to be brittle, and hard to understand and test. The previous suggestion is probably better, all things considered.

Reply:
The third option is to reuse the Quarkus "map non-public field descriptors to method descriptors" trick. The reason this isn't done in the code is because we have a guarantee we won't go through getters/setters, although since our test classes have simple getters/setters, we can use them in the test (which leads to the question: should we have another domain access type: GIZMO_RELAXED or GIZMO_DEFFERED etc. which is GIZMO except of instead of failing on non-public fields, it will use getters/setters of the fields)